x-pack/plugins/maps/public/classes/sources/mvt_single_layer_vector_source/mvt_single_layer_source_settings.tsx

```diff
@@ -0,0 +1,218 @@
+      currentMaxSourceZoom: nextProps.maxSourceZoom,
+      currentFields: clonedFields,
+      previousLayerName: nextProps.layerName,
+      previousMinSourceZoom: nextProps.minSourceZoom,
+      previousMaxSourceZoom: nextProps.maxSourceZoom,
+      previousFields: clonedFields,
+    };
+  }
+
+  _handleChange = _.debounce(() => {
+    this.props.handleChange({
+      layerName: this.state.currentLayerName,
+      minSourceZoom: this.state.currentMinSourceZoom,
+      maxSourceZoom: this.state.currentMaxSourceZoom,
+      fields: this.state.currentFields,
+    });
+  }, 200);
+
+  _handleLayerNameInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const layerName = e.target.value;
+    if (layerName === this.state.currentLayerName) {
+      return;
+    }
+    this.setState({ currentLayerName: layerName }, this._handleChange);
+  };
+
+  _handleFieldChange = (fields: MVTFieldDescriptor[]) => {
+    if (_.isEqual(this.state.currentFields, fields)) {
+      return;
+    }
+    this.setState({ currentFields: fields }, this._handleChange);
+  };
+
+  _handleZoomRangeChange = (e: Value) => {
+    const minSourceZoom = parseInt(e[0] as string, 10);
+    const maxSourceZoom = parseInt(e[1] as string, 10);
+    if (
+      this.state.currentMinSourceZoom === minSourceZoom &&
+      this.state.currentMaxSourceZoom === maxSourceZoom
+    ) {
+      return;
+    }
+    this.setState(
+      { currentMinSourceZoom: minSourceZoom, currentMaxSourceZoom: maxSourceZoom },
+      this._handleChange
+    );
+  };
+
+  render() {
+    const fieldEditor =
+      this.props.includeFields && this.state.currentLayerName !== '' ? (
+        <EuiFormRow
+          label={i18n.translate(
+            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.fieldsMessage',
+            {
+              defaultMessage: 'Fields',
+            }
+          )}
+          helpText={i18n.translate(
+            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.fieldsHelpMessage',
+            {
+              defaultMessage:
+                'Fields which are available in the `{layer}` layer. These can be used for tooltips and dynamic styling.',
+              values: {
+                layer: this.state.currentLayerName,
+              },
+            }
+          )}
+        >
+          <MVTFieldConfigEditor
+            fields={this.state.currentFields.slice()}
+            onChange={this._handleFieldChange}
+          />
+        </EuiFormRow>
+      ) : null;
+
+    return (
+      <Fragment>
+        <EuiFormRow
+          label={i18n.translate(
+            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.layerNameMessage',
+            {
+              defaultMessage: 'Layer name',
+            }
+          )}
+          helpText={i18n.translate(
+            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.layerNameHelpMessage',
+            {
+              defaultMessage: 'Name of the target data layer in the tile',
```

Comment:
There is a lot of help text in this form and it looks visually very different from the other forms on the page. Is all the help text really needed?

Reply:
Reverting to `Tile layer` instead